move_vm/events.md

```diff
@@ -23,23 +23,117 @@ For smart contracts related transactions, there are reserved types for events, s
     * **major\_status** - the major status of error, integer.
     * **sub\_status** - the sub status of error, integer, optional.
     * **message** - text message, optional.
-* **contract\_events** -  contains events generated during smart contract execution. In our example of the script, we generated such an event using EventHandler. Contains next attributes:
-  * **guid** - the unique id of the event.
-  * **sequence\_number** - sequence number of this event within current EvenHandler.
-  * **type** - contains data type, similar to contract arguments, but also could be a struct. A struct could be decoded using [lcs](https://github.com/the729/lcs).
-  * **data** - [lcs](https://github.com/the729/lcs) encoded data in hex \(with prefix - "0x"\). It could be decoded using [lcs](https://github.com/the729/lcs).
+* **contract\_events** -  contains events generated during smart contract execution. In our example of the script, we generated such an event using `Event::emit`. Contains next attributes \(attributes always sorted in the same sequence\):
+  * **sender_address** - address of account which sent transaction that sent event.
+  * **source** - the place where the event was sent, it could be `script` (in case it sent from user script), or path to module which sent event, e.g.: `0x1::Account`.
+  * **type** - contains data type, similar to contract arguments, but also could be a struct \(in such case there will be reference to which indeed struct used in the event\). A struct could be decoded using [lcs](https://github.com/the729/lcs).
+  * **data** - [lcs](https://github.com/the729/lcs) encoded data in hex. It could be decoded using [lcs](https://github.com/the729/lcs).
 
 The **data** field always using LCS encoding \(Libra Canonical Serialization\). There is a community [description](https://github.com/librastartup/libra-canonical-serialization/blob/master/DOCUMENTATION.md) of how it works. Also, Golang [library](https://github.com/the729/lcs), where you can see examples and use for your own projects. So decoding of the **"data"** field should happen with LCS.
 
-There are two reserved events: sent and received events that fire when withdrawing or depositing of resources happens. The implementation you can found in the standard library in [Account](https://github.com/dfinance/dvm/blob/bf457b3145c5e448ece3258bbf67c22326559a12/lang/stdlib/account.move#L14) module.
+There are two reserved events: sent and received events that fire when withdrawing or depositing of resources happens. The implementation you can found in the standard library in [Account](https://github.com/dfinance/dvm/release-v0.4.0/lang/stdlib/account.move) module.
```

Comment:
release-v0.4.0???

Check links twice. It's better to point at master instead of specific release.